File: Day-12/Q_48.c
//Write a program to write function for perfect number.
#include<stdio.h>

int perfect(int n)
{
    int sum=0;
    for(int i=1;i<=n;i++)
    {
        if(n%i==0 && i!=n)
        {
            sum=sum+i;
        }
    }

    if(sum==n)
        {
            return 1;
        }

    return 0;
}
int main()
{
    int n;

    printf("Enter the number: ");
    scanf("%d",&n);

    if(perfect(n))
    {
        printf("Perfect number");
    }

    else
    {
        printf("Not a perfect number");
    }

    return 0;
}

File: Day 12/Q48.java
import java.util.*;
public class Q48
{
    public static boolean perfect(int n)
    {
        if(n<=1)
        {
            return false;
        }
        int sum=1;
        for(int i=2;i<=n/2;i++)
        {
            if(n%i==0)
            {
                sum=sum+i;
            }
        }
        return sum==n;
    }
    public static void main()
    {
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter an integer to check = ");
        int num=sc.nextInt();
        if (perfect(num))
        {
            System.out.println(num+" is a perfect number ");
        }
        else
        {
            System.out.println(num+" is not a perfect number");
        }
    }
}